.github/workflows/build-cmake.yaml

```diff
@@ -0,0 +1,208 @@
+name: Build CMake
+
+on:
+  workflow_dispatch: # Manual trigger
+  push:
+    branches:
+      - master
+  pull_request:
+    paths-ignore:
+      - .gitignore
+      - .github/**
+      - '!.github/workflows/build-cmake.yaml'  # Run check on self change
+      - CONTRIBUTORS
+      - LICENSE
+      - NOTICE
+      - README.md
+      - android/**
+      - iphone/**
+      - data/strings/**
+      - docs/**
+      - packaging/**
+      - pyhelpers/**
+      - tools/**
+      - '!tools/python/test_server/**'
+      - xcode/**
+
+# Cancels previous jobs if the same branch or PR was updated again.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        cmake-build-type: [ Debug, RelWithDebInfo ]
+        environment:
+          - os: ubuntu-24.04
+            cc: clang-18
+            cxx: clang++-18
+            disable-unity-build: ON
+            run-coverage: OFF
+
+          - os: ubuntu-24.04
+            cc: clang-18
+            cxx: clang++-18
+            disable-unity-build: OFF
+            run-coverage: OFF
+
+          - os: ubuntu-24.04
+            cc: clang-18
+            cxx: clang++-18
+            disable-unity-build: OFF
+            run-coverage: ON
+
+          - os: ubuntu-24.04
+            cc: gcc-14
+            cxx: g++-14
+            disable-unity-build: OFF
+            run-coverage: OFF
+
+          - os: macos-15
+            cc: clang
+            cxx: clang++
+            disable-unity-build: OFF
+            run-coverage: OFF
+            developer-dir: /Applications/Xcode_16.app/Contents/Developer
+        exclude:
+          - environment: { run-coverage: ON }
+            cmake-build-type: RelWithDebInfo
+
+    name: "${{ matrix.environment.os }} ${{ matrix.environment.cc }} ${{ matrix.cmake-build-type }}${{ matrix.environment.disable-unity-build == 'ON' && ' No Unity' || '' }}${{ matrix.environment.run-coverage == 'ON' && ' Coverage' || '' }}"
+
+    runs-on: ${{ matrix.environment.os }}
+
+    env:
+      BUILD_DIR: build
+      DEVELOPER_DIR: ${{ matrix.environment.developer-dir }}
+
+    steps:
+      - name: Free disk space by removing .NET, Android and Haskell
+        if: ${{ runner.os == 'Linux' }}
+        shell: bash
+        run: sudo rm -rf /usr/share/dotnet /usr/local/lib/android /opt/ghc
+
+      - name: Checkout sources
+        uses: actions/checkout@v4
```

Comment:
Is version.sh used for cmake builds (then history 200 is needed) or only for ios/android?

Reply:
Not within the scope of this PR. Created an issue: https://github.com/organicmaps/organicmaps/issues/10616